be/src/storage/rowset/index_page.cpp

```diff
@@ -87,55 +87,46 @@ Status IndexPageReader::parse(const Slice& body, const IndexPageFooterPB& footer
 }
 ///////////////////////////////////////////////////////////////////////////////
 
+// This function has the meaning of interval, in fact, it is to find first possible Page which <=search key
 Status IndexPageIterator::seek_at_or_before(const Slice& search_key) {
-    int32_t left = 0;
-    int32_t right = _reader->count() - 1;
-    while (left <= right) {
-        int32_t mid = left + (right - left) / 2;
-        int cmp = search_key.compare(_reader->get_key(mid));
-        if (cmp < 0) {
-            right = mid - 1;
-        } else if (cmp > 0) {
-            left = mid + 1;
-        } else {
-            _pos = mid;
-            return Status::OK();
-        }
-    }
+    _seek(search_key);
+
     // no exact match, the insertion point is `left`
-    if (left == 0) {
+    if (_pos == 0 && search_key.compare(_reader->get_key(0)) < 0) {
         // search key is smaller than all keys
         return Status::NotFound("no page contains the given key");
     }
-    // index entry records the first key of the indexed page,
-    // therefore the first page with keys >= searched key is the one before the insertion point
-    _pos = left - 1;
     return Status::OK();
 }
 
+// This function has the meaning of interval, in fact, it is to find first possible Page which >=search key
 Status IndexPageIterator::seek_at_or_after(const Slice& search_key) {
+    _seek(search_key);
+    return Status::OK();
+}
+
+void IndexPageIterator::_seek(const Slice& search_key) {
```

Comment:
How about renaming it something like `_xxx_lower_bound`, since this logic is similar as `std::lower_bound`?